sshd-scp/src/test/java/org/apache/sshd/client/scp/ScpTest.java

```diff
@@ -589,6 +589,53 @@ public void testScpNativeOnDirWithPattern() throws Exception {
         }
     }
 
+    @Test
+    public void testScpVirtualOnDirWithPattern() throws Exception {
+        Path remoteDir = Files.createTempDirectory("remote");
+        sshd.setFileSystemFactory(new VirtualFileSystemFactory(remoteDir));
```

Comment:
Please note that `sshd` is a **shared** instance - therefore you need to restore it to its original state:
```java
@Test
public void testScpVirtualOnDirWithPattern() throws Exception {
    try {
      ...
       sshd.setFileSystemFactory(...test factory...);
      ...
    } finally {
        sshd.setFileSystemFactory(fileSystemFactory);    // the one created in the test constructor
    }
}
```

Reply:
This is being taken care in `Before` block.
``` 
   @Before
    public void setUp() throws Exception {
        sshd.setFileSystemFactory(fileSystemFactory);
    }
```